sympy/combinatorics/homomorphisms.py

```diff
@@ -437,10 +437,19 @@ def group_isomorphism(G, H):
     First, the set of generators are mapped with the elements of `H` and we check if the mapping induces an isomorphism.
 
     '''
-    # Two infinte FpGroups with the same generators are isomorphic when the relators are same but are ordered differently.
+
     if isinstance(G, FpGroup) and isinstance(H, FpGroup):
+        G = simplify_presentation(G)
+        H = simplify_presentation(H)
+        # Two infinite FpGroups with the same generators are isomorphic
+        # when the relators are same but are ordered differently.
         if G.generators == H.generators and (G.relators).sort() == (H.relators).sort():
             return homomorphism(G, H, G.generators, H.generators)
+        # Two infinite FpGroups with the same generators are isomorphic
+        # if there exists a map from the generators of one group to the other.
+        gen_map = homomorphism(G, H, G.generators, H.generators)
+        if gen_map:
+            return gen_map
```

Comment:
That is not true. There is always a homomorphism from a free group to a finitely presented groups that is its quotient, but this is not an isomorphism.

If I understand what you tried to do here correctly, you don't need these lines at all - it already works without, e.g. the test you commented on as `#Trivial case`.

What we do need is something like this:
```
>>> F, a, b = free_group("a b")
>>> G = FpGroup(F, [a**2])
>>> H = FpGroup(F, [b**2])
>>> is_isomorphic(G, H)
True
```